specification/introduction/introduction.tex

```diff
@@ -3,59 +3,81 @@ \chapter{Introduction}\label{sec:introduction}
+the specification.
 
 \section{Design principles}
 
-UAVCAN is designed to adhere to the following set of basic principles.
-
 \begin{description}
     \item[Democratic network] --- There will be no master node.
     All nodes in the network will have the same communication rights; there must be no single point of failure.
 
-    \item[Nodes can exchange long payloads] --- Nodes must be provided with a simple way of exchanging large
-    data structures that cannot fit into a single transport frame (such as GNSS solutions, 3D vectors, etc.).
-    UAVCAN should perform automatic transfer decomposition and reassembly at the protocol level,
+    \item[Facilitation of functional safety] --- A system designer relying on UAVCAN will have the necessary
+    guarantees and tools at their disposal to analyze the system and ensure its correct behavior.
+
+    \item[High-level communication abstractions] --- The protocol will support publish/subscribe and remote procedure
+    call communication semantics with statically defined and statically verified data types (schema).
+    The data types used for communication will be defined in a clear, platform-agnostic way
+    that can be easily understood by machines, including humans.
+
+    \item[Facilitation of cross-vendor interoperability] --- UAVCAN will be a common foundation that
+    different vendors can build upon to ensure interoperability of their equipment.
+    UAVCAN will provide a generic set of standard application-agnostic communication data types.
+
+    \item[Well-defined generic high-level functions] --- UAVCAN will define standard services
+    and messages for common high-level functions, such as network discovery, node configuration,
+    node software update, node status monitoring\footnote{Which naturally grows into a vehicle-wide
+    health monitoring solution.}, network-wide time synchronization, plug-and-play node support, etc.
+
+    \item[Atomic data abstractions] --- Nodes must be provided with a simple way of exchanging large
+    data structures that exceed the capacity of a single transport frame\footnote{A \emph{transport frame} is
+    an atomic transmission unit defined by the underlying transport protocol. For example, a CAN frame.}.
+    UAVCAN should perform automatic data decomposition and reassembly at the protocol level,
     hiding the related complexity from the application.
 
+    \item[High throughput, low latency, determinism] --- UAVCAN must be suitable for use in
+    hard real-time deterministic systems.
+
     \item[Support for redundant interfaces and redundant nodes] --- UAVCAN must be suitable for use in
     applications that require modular redundancy.
 
-    \item[High throughput, low latency communication] --- UAVCAN must be suitable for use in
-    hard real-time systems.
-
     \item[Simple logic, low computational requirements] --- UAVCAN targets a wide variety of embedded systems,
```

Comment:
Consider this a form of marketing. A human engineer would open the doc and read through the list of goals while evaluating each entry against their own problems they have to solve. I think it is important to make a very obvious statement that we do, in fact, care about memory impaired microcontrollers because the preceding entries might lead one to believe otherwise.

Reply:
My comment was meant as an addition. Also meant as "marketing". Stating that supporting the minimal set of the protocol will be inexpensive in terms of engineering hours and change required, and more advanced support can be incrementally added in a pick and choose fashion as more features is desired.

Reply:
Fixed?

Reply:
:+1: